test/SILOptimizer/simplify_cfg.sil

```diff
@@ -2629,3 +2629,35 @@ bb3(%4 : $Optional<Int>):
 }
 
 
+// CHECK-LABEL: sil @dont_crash_on_enum_payload_is_enum
```

Comment:
Can we add a comment that indicates what is the purpose of this test?  The only indicator is this `dont_crash_on_enum_payload_is_enum` label, but that is a bit opaque.  Referencing the SR or Radar # would be good too.